pgml-dashboard/src/lib.rs

```diff
@@ -626,3 +660,93 @@ pub fn routes() -> Vec<Route> {
 pub async fn migrate(pool: &PgPool) -> anyhow::Result<()> {
     Ok(sqlx::migrate!("./migrations").run(pool).await?)
 }
+
+#[cfg(test)]
+mod test_current_user {
+    use crate::CurrentUser;
+    use crate::models::User;
+    use std::collections::HashMap;
+
+    // New current user should set user and visible clusters to none.
+    #[test]
+    fn test_new_current_user() {
+        let test_user = CurrentUser::new();
+        assert!(test_user.get_user().is_none() && test_user.get_visible_clusters().is_none());
+    }
+
+    // It should be possible to set and get current user. 
+    #[test]
+    fn test_set_get_current_user() {
+        let test_user = CurrentUser::new();
+        let new_user_data = User {
+            id: 1, 
+            email: "test@email.com".to_string(),
+        };
+
+        test_user.set_user(new_user_data.clone());
+        assert_eq!(
+        (
+            test_user.get_user().unwrap().id, 
+            test_user.get_user().unwrap().email,
+        ), (
+            new_user_data.id,
+            new_user_data.email,
+        ));
+    }
+
+    // It should be possible to set and get visible clusters. 
+    #[test]
+    fn test_set_get_visible_clusters() {
+        let test_user = CurrentUser::new();
+        let new_visible_clusters: HashMap<String, String> = HashMap::from([
+            ("1".to_string(), "123456789".to_string()),
+            ("2".to_string(), "987654321".to_string()),
+        ]);
+        test_user.set_visible_clusters(new_visible_clusters.clone());
+        assert_eq!(
+            (
+                test_user.get_visible_clusters().unwrap().get("1"), 
+                test_user.get_visible_clusters().unwrap().get("2"),
+            ), (
+                new_visible_clusters.get("1"), 
+                new_visible_clusters.get("2")
+            )
+        )
+    }
+
+    // It should reset current user and visible clusters to none. 
+    #[test]
+    fn test_reset_user_to_default() {
+        let test_user = CurrentUser::new();
```

Comment:
This is single-threaded, single-request simulation. I think you'll be able to validate that the current implementation doesn't work if you spawn 1000 threads and make them set and fetch the current user very quickly in succession.